[Glenn Lim Jun Wei] iP

glennljw opened the pull request:
# Duke

> “The true price of anything you do is the amount of time you exchange for it.” – Henry David Thoreau ([source](https://dansilvestre.com/productivity-quotes/))

With Duke, the amount of time you spend on keeping track of your tasks would be significantly reduced. Thus the true price _(expense)_ of that can increase your productivity in other aspects which are regarded to be of higher significance and importance.

## ***Why Duke?***
Duke can up the level of task management.
- [x] ~~Easy~~ EXTREMELY EASY to use
- [x] Multiple **fuss-free** features
- [x] More upcoming features

Here are the features that Duke currently provides:

* Manage tasks
* Manage deadlines
* Manage events
* Find your tasks
* Saves your tasks

## ***How to download Duke?***

1. Simply download the ***ip.jar*** file from this [link](https://github.com/glennljw/ip/releases).
2. Run the ***ip.jar*** file.
3. Add your tasks.
4. Let Duke remember your tasks for you! :wink:

##

Here's a snippet of my main method for you to try implementing your own Duke!
```java
public class Duke {
    public static void main(String[] args) {
        Duke duke = new Duke();
        duke.run();
    }
}
```

Commit:
Add Gradle support

glennljw committed:
Level-1: Greet, Echo, Exit

glennljw committed:
Level-2

glennljw committed:
Level-3: Mark as Done

glennljw committed:
Level-4: ToDo, Event, Deadline

glennljw committed:
Deleting .class files according to .gitignore

glennljw committed:
A-TextUiTesting: Automated Text UI Testing

glennljw committed:
Level-5: Handle Errors

glennljw committed:
Level-6: Delete

glennljw committed:
Added comments for Level-6

glennljw committed:
A-Enums: Use Enums

glennljw committed:
Update code to follow Style Guide according to CS2103T module

glennljw committed:
Clean up code to make switch statements flow more smoothly

glennljw committed:
Add JavaDocs

glennljw committed:
Update code to not exceed 120 chars per line

glennljw committed:
Add packages and update bash script accordingly

glennljw committed:
Level-7: Save

glennljw committed:
Level-8: Dates and Times

glennljw committed:
Update code to resolve merge conflict

glennljw committed:
Fix bugs in code and resolve merge conflicts

glennljw committed:
Abstract save file management into separate Class

glennljw committed:
Fix bugs

glennljw committed:
A-MoreOOP: Use More OOP

glennljw committed:
A-MoreOOP: Use More OOP

glennljw committed:
A-Packages: Organize into Packages

glennljw committed:
A-JUnit: Add JUnit Tests

glennljw committed:
A-JUnit: Add JUnit Tests

glennljw committed:
A-JavaDoc: JavaDoc

glennljw committed:
A-CodingStandard: Follow the Coding Standard

glennljw committed:
Level-9: Find

glennljw committed:
Merge branch 'branch-Level-9'

glennljw committed:
Merge branch 'branch-A-CodingStandard'

glennljw committed:
Merge branch 'branch-A-JavaDoc'

glennljw committed:
Add JavaDocs for FindCommand class

glennljw committed:
Refactor code and add JavaDocs

glennljw committed:
Add package for utility classes

glennljw committed:
Fix wildcard import issue due to IntelliJ code style

glennljw committed:
Fix bugs for printing Event and Deadline

glennljw committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

glennljw committed:
A-Gradle: Use Gradle, A-CheckStyle: Use CheckStyle

glennljw committed:
A-CheckStyle: Use CheckStyle

glennljw committed:
Merge branch 'branch-A-Gradle'

glennljw committed:
Level-10: GUI

glennljw committed:
Merge branch 'branch-Level-10'

glennljw committed:
Update JavaFX GUI

glennljw committed:
Add welcome message to GUI

soloplxya commented on src/main/java/duke/util/Save.java:
```diff
@@ -0,0 +1,244 @@
+/**
+ * Represents a handler to load a save file, or save to a file.
+ */
+public class Save {
```
Perhaps can consider an alternative name for the class since the class also contains methods other than saving. Other than that, the code looks pretty good.

thewrik replied:
Echoing what @soloplxya suggested on this.

soloplxya commented on src/main/java/duke/util/Save.java:
```diff
@@ -0,0 +1,244 @@
+     *
+     * @param taskList ArrayList of Strings to be processed
+     */
+    public void process(ArrayList<String> taskList) throws InvalidCommandException {
```
Perhaps can consider using "fileTasks" instead of taskList for the param name.

soloplxya commented on src/main/java/duke/util/Save.java:
```diff
@@ -0,0 +1,244 @@
+ * Represents a handler to load a save file, or save to a file.
+ */
+public class Save {
+    private static final String DUKE_PATHNAME = "src/main/java/Duke/data/duke.txt";
```
Good job for uppercasing the constant names. 👍

soloplxya reviewed (commented):
Overall, I like the structure of your code. The coding standard has also been well-adhered to. Good job!

thewrik reviewed (commented):
Appreciate that you took the initiative to structure each command as a Class.

thewrik commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,60 @@
+        ArrayList<Boolean> keywordMatchedList = new ArrayList<>(tasks.getCount());
+        boolean hasMatch = false;
+
+        taskArrList.forEach(x -> {
```
Nice, sleek writing.

thewrik commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,41 @@
+                    + " " + tasks.getTask(taskNum);
+            response += markedString;
+        } catch (IndexOutOfBoundsException e) {
+            response = "☹ Woof Woof!!! This task cannot be found with my Wonka eyes!!!";
```
Haha, great prompts.

thewrik commented on src/main/java/duke/data/duke.txt:
```diff
@@ -0,0 +1,4 @@
+T / 1 / read book
```
Feel like this file could be added to .gitignore

thewrik commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,101 @@
+ * Represents a deadline to be fulfilled. This deadline is a task that possesses a state/status that is by default
+ * not done.
+ */
+public class Deadline extends Task {
```
`public class DeadlineTask extends Task {` is probably better, given the Naming standards for inherited classes. (You used it in `Command`, yourself.)

thewrik reviewed (commented):
Appreciate that you took the initiative to structure each command as a Class. Generally clean code, with some minor nitpicks.

glennljw committed:
Add unit test for AddCommand

glennljw committed:
Change GUI of Duke

glennljw committed:
Add bye behaviour for GUI

glennljw committed:
Fix save behaviour to save on each alteration to task list

glennljw committed:
Remove unnecessary comments

glennljw committed:
Change build.gradle

glennljw committed:
Refactor code

glennljw committed:
A-Assertions: Use Assertions

Assertions not included

Assertions are used to define assumptions about the program state so
that the runtime can verify them. By adding assertions, the developer
can identify if the result is as intended, and can detect possible bugs
in the code in the event of an assertion failure.

Let's add insertions to check the intended behaviour of the task list
after adding, deleting, marking and unmarking commands.

Ensure that the task list will be properly modified upon the execution
of commands.

glennljw committed:
Refactor code: SLAP

glennljw committed:
Refactor code: Avoid long methods

glennljw committed:
Merge pull request #1 from glennljw/branch-A-CodeQuality

Branch A-CodeQuality

glennljw committed:
Resolve merge conflict

glennljw committed:
Merge pull request #2 from glennljw/branch-A-Assertions

Branch A-Assertions

glennljw committed:
Refactor code in Duke

glennljw committed:
A-CI: Set up CI

glennljw committed:
B-FixedDurationTasks: Unscheduled tasks with a fixed duration

glennljw committed:
Change unit tests

glennljw committed:
Add unit tests for TaskList

glennljw committed:
A-BetterGui: Better GUI

glennljw committed:
Fix minor bugs in chatbot behaviour

glennljw committed:
Remove empty default block

glennljw committed:
A-UserGuide: User Guide

glennljw committed:
Update User Guide

glennljw committed:
Set theme jekyll-theme-slate

glennljw committed:
Fix save file path

glennljw committed:
Merge branch 'master' of https://github.com/glennljw/ip

glennljw committed:
Add Wonka app icon

glennljw committed:
Update Gui

glennljw committed:
Update Ui.png

glennljw committed:
Update README.md to reflect correct url for image

glennljw committed:
Remove resizable property of chatbot

glennljw committed:
Change response behaviour

glennljw committed:
Change Ui.png

glennljw committed:
Fix README.md typo

glennljw committed:
Change save file path

glennljw committed:
Update README.md to include examples for todo
